src/components/login-page/AuthProvider.jsx

```diff
@@ -0,0 +1,53 @@
+import { AuthContext } from '../../utils/constants';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const AuthProvider = ({ children }) => {
+    const [user, setUser] = useState(null);
+    const [token, setToken] = useState(localStorage.getItem('power') || '');
+    const navigate = useNavigate();
+
+    const loginAction = async (userAndPass) => {

```

Comment:
the loginAction function doesn't really return anything no matter the result. one thing we can improve is to have it return something like a boolean true/false or an object like { loginSuccessful: true/false }.
That way, when it is used, there is a chance to tell if it worked or not and react according to it.